watchtower/wtdb/session_id.go

```diff
@@ -20,6 +22,29 @@ func NewSessionIDFromPubKey(pubKey *btcec.PublicKey) SessionID {
 	return sid
 }
 
+// NonceFromSeqNum creates a deterministic nonce for a particular (sessionid,
+// sequence number) pair. The nonce is derived by taking the last 10 bytes of
+// the session id, concatenated with the 2 byte, big-endian sequence number.
+//
+// The nonce is derived in this manner to protect confidentiality of the
+// payloads when backing up state updates when multiple backups will be made.
+// Since the encryption key will always be equal to the txid, we must take care
+// not to reuse the same nonce when performing multiple backups of the same
```

Comment:
Nice! Wasn't aware that the 192-bit variant was exposed, I agree the randomization is the safer approach.

Reply:
with the existing nonce generation, the protections would be to use truly ephemeral keys (instead of HD keys) and to persist an encrypted update under for the session/seqnum so it can be resent on startup. arguably the first should still be done with a randomized nonce, though it does simplify the latter at the expense of tower storage